documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,234 @@
+* Data don’t save in data base, then list of departments records is displaying to administrator.
+* If the administrator selects the menu item "Department ” or "Employee", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](edit_department.jpg)
+<br>Pic. 2.3 Edit department.
+
+<b>When editing a order, the following details are entered:</b>
+* Name department - name department
+* Add employee - adding department employees(optional field)
+
+### Removing the department
+
+<b>Main scenario:</b>
+* Administrator, while in the list of departments presses the "Delete" button in the selected department line;
+* If the department can be removed, a confirmation dialog is displayed;
+* Administrator confirms the removal of the order;
+* Record is deleted from database;
+* If error occurs, then error message displays;
+* If department record is successfully deleted, then list of departments without deleted records is displaying.
+
+![](del_department.jpg)
+<br>Pic. 2.4 Delete department dialog.
+
+<b>Cancel operation scenario:</b>
+* Administrator is in display mode of department list and press “Delete” button;
+* Application displays confirmation dialog “Please confirm delete department”;
+* Administrator press “Cancel” button;
+* List of departments without changes is displaying. 
+<!--/DEPARTMENT-->
+
+<!--EMPLOYEE-->
+## Employee
+### Display list of employee
+
+This mode is intended for viewing and editing the employees list
+
+<b>Main scenario:</b>
+* User selects item “Employee”;
+* Application displays list of Employee.
+
+![](employee_listt.jpg)
+Pic. 3.1 View the Employee list.
+
+<b>The list displays the following columns:</b>
+* ID – unique employee number;
+* Name - first name and last name
+* Date of birth - employee's date of birth
+* Department - the department in which the employee consists
+* Positions - employee positions
+* Specialization - skills that the employee has
+* Experience - work experience
+* Wages - employee's salary
+
+<b>Filtering by date:</b>
+* In the employee list view mode, the administrator sets a filter by date and clicks the update list button (to the right of the date input field).;
+* The application will show employees only during a certain period of the date of birth.
+
+<b>Restrictions:</b>
+* Start date of the period should be less then end date of the period;
+* If start date is blank, then filtering by end date only.
+* If end date is blank, then filtering by start date only.
+* Updating data after selecting the filtering conditions is carried out by pressing the “Refresh” button.
+
+### Add employee
+
+<b>Main scenario:</b>
+* Administrator clicks the “Add” button in the employee list view mode;
+* Application displays form to enter employee data;
+* Administrator enters employee data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
```

Comment:
It would be great to see a mockup with error messages.

Reply:
Fixed